source/agora/consensus/PreImage.d

```diff
@@ -538,41 +523,43 @@ public void getCycleSeed (in KeyPair pair, in uint validator_cycle, out Hash see
+        seed = Hash(`0xfa126384ce0da0fc135fa7670fbe9686c7263daaa6cded62cf33d88c94c690d6f6d339aa7fa4da0aecae836d4882de2f951a3f7d247f9c0aa289022997498abf`);
     else if (pair == D)
-        seed = Hash(`0x64d1b6e98019df73b59c3593e2ca1baea53aeeb3cac18748d3ab50473a64963c76c318913158349caf368730b027eb8c4596b743308a7e53a65e484016c8e2ca`);
+        seed = Hash(`0x988430c307a2699a82a3c9f49f719b113d5c669458b9dbc12d51416ae4e4fd62ccad3bc7350a10c2f8bd77e462ea6400d2e6b65a8c35b684dbdd1cb03e7e494d`);
     else if (pair == E)
-        seed = Hash(`0x0bb70cf7d265661d006bc5bfd5bf70a66c5b32e54d05ee37bf023c4c0bf9b3bc76faf38d656c25e254aae7ea73aabba1902192575da80cfe10a24363d3e41118`);
+        seed = Hash(`0x9b65e3277352ed506736179bd3b49dd2577e5279fffbd0dc208d2e70e2534b1e66f74bb76c386056b9db5c8a6f543413ca807b7a0265f34cfebb0ee5950c8745`);
     else if (pair == F)
-        seed = Hash(`0xbb19e8bae8f03ce2aee0c15f83089c87dd208b44d17683ace877696a98680730c4cae9f762366a70dd00885cf0ff2eb228f285885ebda9e6dc11c0288158d8cf`);
+        seed = Hash(`0x4a3e76d46eface58683e647a456e5812de51db4dfbeab261a90418c9b43bf306c36c12b9b6c0f2e541a859ddf2f8d1f7621391ece992d5947ccbe3a86d411326`);
     else if (pair == G)
-        seed = Hash(`0x90f2a3d469831bad2d27f4698c463dad56c8d942b28859e38c2f4b1bc975605f0641f62aff2edf23326b67ad3b34c47bc43f96167485d4db2529d5e84fc7e994`);
+        seed = Hash(`0x18f65d7f34ce57b5a92f5647e6c5190d28e5fb28ef9edd9dec4762f8c7e3efbd01fbe062bbce0640fba4f90d918af93da27a409ef76dc6787741f3a34159dfb6`);
     else if (pair == H)
-        seed = Hash(`0xa84c60d88e1c075d8d70c77426c4724ac8bd38f83030fb16a3db71890396548572031907fbf7adba1034a7f9b7b4f2ae11155010cf677ed81cc32728f6ec7a06`);
+        seed = Hash(`0xec5f4d0538b5f60b4fe1dac8bad6fe5d9527531165fa3ea139d36c61b89eb37bd896a613d878d0f4140a59fae6b1e7ad95381a17bb54089943a39acf4c3b998e`);
     else if (pair == J)
-        seed = Hash(`0xe39ae981988560d89e85f340c087fea46f1d9eefe73400f2a85edd3d3adee2bc5a3efd6e6ce9de04d73a3fe2554a1b1035450b7be9eaade24bbb789585aa0015`);
+        seed = Hash(`0x3256f762cee8e5345361d3db3d500ce7fb1c9591a0f051a264e383f764dc8c41033ce5f094917f58311c8597a450784c92bdfae947ca9c39a4ce62373a8631b4`);
     else if (pair == K)
-        seed = Hash(`0x72d84ce0dfc74f856909a3f49dffdb8b68c39212f1229fddec3fb071c3bff84983645f8caab335189a710502c1020ec4b855f70f8b1493718848aa9dcc761fa7`);
+        seed = Hash(`0xf32b9715a5ba6c8c26ac9dd92179c9e549cdbbe18066c02db7dc6fff62e83b8c3ac4052265b1320af39b93f3a70e2f9b5f860c39d1f2b27a7e66d3554628bcec`);
     else
     {
         auto cycle = PreImageCycle(pair.secret, 20, PreImageCycle.PreImageCount, Height(0));
         seed = cycle[seed_height];
-        throw new Exception(format!"There's no seed for %s. The Hash of (Hash(`%s`) should be added."
+        writeln(format!"There's no seed for %s. The Hash of (Hash(`%s`) should be added."
```

Comment:
I guess the original code is more suitable for our test because we should add the missing seed for the new key after the exception.

Reply:
I think there is no reason to interrupt the test just for missing seeds, logging the seed so that they can be updated when convenient is a better approach imo.